### DISCUSSION TITLE
String is indirectly referenced from required .class files

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-jdt/eclipse.jdt.core#525] [Q&A]
I'm currently working on Tycho to enhance the classpath handling but encountering a problem when running the aggregator to verify it does not creates to much problems for platform.

Now I'm getting the following error
```
Error:  Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:4.0.0-SNAPSHOT:compile (default-compile) on project org.eclipse.jdt.junit.core: Compilation failure: Compilation failure: 
Error:  /home/runner/work/tycho/tycho/platform/eclipse.jdt.ui/org.eclipse.jdt.junit.core/src/org/eclipse/jdt/internal/junit/JunitPreferenceInitializer.java: 
Error:  	/*******************************************************************************
Error:  	^
Error:  Type java.lang.String is indirectly referenced from required .class files but cannot be resolved since the declaring package java.lang exported from module java.base conflicts with a package accessible from module <unnamed>
```

and I don't really understand this, how can `java.lang` come from different modules?

**Comment by iloveeclipse** [comment]
> and I don't really understand this, how can `java.lang` come from different modules?

If some clever guy put some classes into java.lang package and included that in his own jar, it will be in different module...
Grep through all dependencies and check who puts some garbage on class path...
